internal/controller/clusterdeployment_controller.go

```diff
@@ -1308,6 +1309,16 @@ func (r *ClusterDeploymentReconciler) waitClusterDataSourceDeletion(ctx context.
 	cdsKey := client.ObjectKey{Name: scope.cd.Name, Namespace: scope.cd.Namespace}
 	err := r.MgmtClient.Get(ctx, cdsKey, cds)
 	if err == nil { // if NO error
+		cd := scope.cd
+		if err := r.MgmtClient.Delete(ctx, &kcmv1.ClusterDataSource{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      cd.Name,
+				Namespace: cd.Namespace,
+			},
+		}); client.IgnoreNotFound(err) != nil {
+			r.setCondition(cd, kcmv1.DeletingCondition, err)
+			return false, err
+		}
```

Comment:
```suggestion
		if err := r.MgmtClient.Delete(ctx, cds); client.IgnoreNotFound(err) != nil {
			return false, fmt.Errorf("failed to delete ClusterDataSource %s: %w", cdsKey, err)
		}
```

Condition should be set by the caller, and the message should also be amended from `failed to wait ...` to `failed to delete ...`